File: src/main/java/com/futurecollars/invoices/model/Invoice.java
package com.futurecollars.invoices.model;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public class Invoice {

  private UUID id;
  private LocalDate date;
  private Company seller;
  private Company buyer;
  private List<InvoiceEntry> entries;

}

File: angularClient/src/app/invoice.ts
import { Company } from './Company';
import { InvoiceEntry } from './InvoiceEntry';

export class Invoice {
    id: string;
    invoiceNumber: string;
    date: string;

    buyer: Company;
    seller: Company;
    entries: InvoiceEntry[] = new Array();

    constructor() {
      this.buyer = new Company();
      this.seller = new Company();
    }
}
